src/govuk/components/summary-list/template.njk

```diff
@@ -7,38 +7,82 @@
 {% endfor -%}
 
-<dl class="govuk-summary-list {%- if params.classes %} {{ params.classes }}{% endif %}"{% for attribute, value in params.attributes %} {{attribute}}="{{value}}"{% endfor %}>
-  {% for row in params.rows %}
-    {% if row %}
-      <div class="govuk-summary-list__row {%- if anyRowHasActions and not row.actions.items %} govuk-summary-list__row--no-actions{% endif %} {%- if row.classes %} {{ row.classes }}{% endif %}">
-        <dt class="govuk-summary-list__key {%- if row.key.classes %} {{ row.key.classes }}{% endif %}">
-          {{ row.key.html | safe if row.key.html else row.key.text }}
-        </dt>
-        <dd class="govuk-summary-list__value {%- if row.value.classes %} {{ row.value.classes }}{% endif %}">
-          {{ row.value.html | indent(8) | trim | safe if row.value.html else row.value.text }}
-        </dd>
-        {% if row.actions.items.length %}
-          <dd class="govuk-summary-list__actions {%- if row.actions.classes %} {{ row.actions.classes }}{% endif %}">
-            {% if row.actions.items.length == 1 %}
-              {{ _actionLink(row.actions.items[0]) | indent(12) | trim }}
-            {% else %}
-              <ul class="govuk-summary-list__actions-list">
-                {%- for action in row.actions.items -%}
-                  <li class="govuk-summary-list__actions-list-item">
-                    {{- _actionLink(action) | indent(18) | trim -}}
-                  </li>
-                {%- endfor -%}
-              </ul>
-            {% endif %}
+{%- set summaryList -%}
+  <dl class="govuk-summary-list {%- if params.classes %} {{ params.classes }}{% endif %}"{% for attribute, value in params.attributes %} {{attribute}}="{{value}}"{% endfor %}>
+    {% for row in params.rows %}
+      {% if row %}
+        <div class="govuk-summary-list__row {%- if anyRowHasActions and not row.actions.items %} govuk-summary-list__row--no-actions{% endif %} {%- if row.classes %} {{ row.classes }}{% endif %}">
+          <dt class="govuk-summary-list__key {%- if row.key.classes %} {{ row.key.classes }}{% endif %}">
+            {{ row.key.html | safe if row.key.html else row.key.text }}
+          </dt>
+          <dd class="govuk-summary-list__value {%- if row.value.classes %} {{ row.value.classes }}{% endif %}">
+            {{ row.value.html | indent(8) | trim | safe if row.value.html else row.value.text }}
           </dd>
-        {% endif %}
-      </div>
-    {% endif %}
-  {% endfor %}
-</dl>
+          {% if row.actions.items.length %}
+            <dd class="govuk-summary-list__actions {%- if row.actions.classes %} {{ row.actions.classes }}{% endif %}">
+              {% if row.actions.items.length == 1 %}
+                {{ _actionLink(row.actions.items[0]) | indent(12) | trim }}
+              {% else %}
+                <ul class="govuk-summary-list__actions-list">
+                  {%- for action in row.actions.items -%}
```

Comment:
@owenatgov Just a quick check, what happens if there are no actions—an empty `<ul>`?

Reply:
In that case it wouldn't meet the `if` on L21 (length = 0 = false) and not render any part of the action list.

Reply:
Ahh @querkmachine, my eyes lied to me (false alarm). Thanks for checking